### ISSUE TITLE
 .setRelationLoadStrategy("query") missing

### ISSUE THREAD

**Issue by vlrevolution** [typeorm/typeorm#8866]
### What was unclear or otherwise insufficient?

Documentation for 0.3.0 (https://github.com/typeorm/typeorm/pull/8616) states the following:

    Old ways of custom repository creation were deprecated.

    added new option on relation load strategy called relationLoadStrategy.
    Relation load strategy is used on entity load and their relations load when you query entities from the database.
    Used on find* methods and QueryBuilder. Value can be set to join or query.
        join loads relations using SQL's JOIN
        query executes separate SQL queries for each relation
        
    And QueryBuilder:

```typescript
userRepository
    .createQueryBuilder()
    .setRelationLoadStrategy("query")
```
However I cannot locate any reference to function 'setRelationLoadStrategy' in the 0.3.0 codebase. Is this a mistake and is not implemented?

**Comment by pleerock** [comment]
Yes looks like an issue. Feel free to contribute! Just add a new method to SelectQueryBuilder called `setRelationLoadStrategy` and set given strategy to the expressionMap.

**Comment by pleerock** [comment]
oh, I see. Current implementation works only when using `FindOptions`. We have to make much more changes if we want this to be supported by QueryBuilder as well.

**Comment by vlrevolution** [comment]
Use this syntax, guys (sucks that can't use querybuilder with it but this helped me do like a 100x increase in performance vs joins with 5 joins):
```typescript
    let results = await this.repo.find({
      relations: {
        relationName: true, // Setting the relationName key to true in relations object forces to use query!
      },
    });
```

**Comment by Ginden** [comment]
>  tbh most of the stuff that querybuilder enables can be now achieved with .find too 

Lately I was thinking about adding find operators `And` and `Or`, adding multicolumn `Raw` to get ~99% of QueryBuilder functionality in `find`.

**Comment by hienngm** [comment]
@Rainson12 My workaround is using `.setFindOptions` to load the relations which need to use the query strategy. And chain it with join, select, etc. to perform a more complicated query.
ex: 
<img width="400" alt="image (6) (1)" src="https://user-images.githubusercontent.com/112270783/194884721-ad75ed44-9959-4941-a6fd-cfbdf35f1ab5.png">
I think it makes sense because It will be very confusing when we are using the keyword “join” (leftJoin or innerJoin) and somehow Typeorm changes it to seperate queries with `.setRelationLoadStrategy `.

**Comment by depyronick** [comment]
> @Rainson12 My workaround is using `.setFindOptions` to load the relations which need to use the query strategy. And chain it with join, select, etc. to perform a more complicated query. ex: <img alt="image (6) (1)" width="400" src="https://user-images.githubusercontent.com/112270783/194884721-ad75ed44-9959-4941-a6fd-cfbdf35f1ab5.png"> I think it makes sense because It will be very confusing when we are using the keyword “join” (leftJoin or innerJoin) and somehow Typeorm changes it to seperate queries with `.setRelationLoadStrategy `.

when using this workaround, you cannot do where on relation fields

**Comment by depyronick** [comment]
> Actually, there are two ways to accomplish this. The first option is to pass the "where" condition into `.setFindOption`, like this:
> 
> ```ts
>   this.createQueryBuilder("post")
>       .setFindOptions({
>         relations: { experience: true, discount: { type: true } },
>         relationLoadStrategy: "query",
>         where: { discount: { type: { id: "1" } } },
>       })
> ```
> 
> Alternatively, you could use a join and then chain it with `.where`, like this:
> 
> ```ts
>   this.createQueryBuilder("post")
>       .setFindOptions({
>         relations: { experience: true, discount: { type: true } },
>         relationLoadStrategy: "query",
>         // where: { discount: { type: { id: "1" } } },
>       })
>       .innerJoin("post.discount", "discount")
>       .innerJoin("discount.type", "discountType")
>       .where("discountType.id = 1")
> ```
> 
> Both methods produce the same behavior as using `relationLoadStrategy` in Repository. I hope you find it useful.

In my use case, there are several `andWhere` and `orWhere`s. I could not make it with the second approach. Since I am in need of use of `Brackets`, the first approach is not suitable for me.

The problem with the second approach is that because it does not join the tables, relation alias is not defined.

**Comment by scr4bble** [comment]
So what is the recommended solution for using the 'query' relationLoadStrategy in querybuilder?
We are having issues with relationLoadStrategy when being used by find*() methods so if we could switch to querybuilder and use the queried relations there, it would be great.
https://github.com/typeorm/typeorm/issues/4738

**Comment by reactuz** [comment]
@pleerock do you have any plans to implement this in near future?

**Comment by spanwair** [comment]
I guess you forgot to put it to schema.prisma file:
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["relationJoins"]
}
